lib/ansible/executor/action_write_locks.py

```diff
@@ -15,30 +15,31 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-# Make coding more python3-ish
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
 from multiprocessing import Lock
+from multiprocessing import Manager
 
 from ansible.module_utils.facts.system.pkg_mgr import PKG_MGRS
 
-if 'action_write_locks' not in globals():
-    # Do not initialize this more than once because it seems to bash
-    # the existing one.  multiprocessing must be reloading the module
-    # when it forks?
-    action_write_locks = dict()
+try:
+    # Do not initialize this more than once.
+    action_write_locks
+except NameError:
+    # These plugins are known to be called directly by action plugins
+    # with names differing from the action plugin name. We precreate
+    # them here as an optimization.
+    action_write_locks = Manager().dict()
+
+    for i in ["copy", "file", "setup", "slurp", "stat"]:
```

Comment:
this probably needs updating to ansible.legacy.X see constants.py for function that deals with the multiple possible names

Reply:
This was derived from this same module. Should it continue to use the assumed names [here](https://github.com/ansible/ansible/pull/74895/files/71d6c087c22b763b84eb7396d4e4dbb44adb1b49#diff-b356a647a1cfc0932905abde09f8c03b2686464dd6036699a4b01aae7f62bc11L42-L44)?

Reply:
This function https://github.com/ansible/ansible/blob/devel/lib/ansible/constants.py#L43-L45 ?

Reply:
https://github.com/ansible/ansible/blob/devel/lib/ansible/constants.py#L66 <= see how these constants are defined, this line (which did exist already) was missed during that conversion, which means this optimization is mostly being skipped/ignored.

Reply:
we probably want to move this 'adhoc list' into constants.py anyways

Reply:
Done. https://github.com/ansible/ansible/pull/74895/files#diff-7285a4bf3f000f5acb1faa7869cf1ae5fa788ab69abec608eff91133d07495ccR66-R80 - The list of actions has been added to the constants.